vault/resource_kv_secret_v2.go

```diff
@@ -184,13 +261,72 @@ func kvSecretV2Read(_ context.Context, d *schema.ResourceData, meta interface{})
 				if err := d.Set(consts.FieldMetadata, serializeDataMapToString(v)); err != nil {
 					return diag.FromErr(err)
 				}
+
+				// Read & Set custom metadata
+				if _, ok := v[consts.FieldCustomMetadata]; ok {
+					cm, err := readKVV2Metadata(d, client)
+					if err != nil {
+						return diag.FromErr(err)
+					}
+
+					customMetadata := []interface{}{cm}
```

Comment:
nit: variable is only used once, you might want to inline the value in the call to `d.Set`, below.